src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -760,6 +760,74 @@ def test_fw_gen_vardecl_visibility(fortran_writer):
                       "end type var\n")
 
 
+def test_fw_gen_proceduredecl(fortran_writer):
```

Comment:
Should this test be part of `fortran_gen_decls_test.py` rather than in this file?

Reply:
Most of `gen_vardecl`, `gen_typedecl`, etc. tests are in this file, so I did the same. But indeed, they should probably all be moved there.

Reply:
Yes, we're gradually splitting things up but since moving *and* editing things at the same time can be hard to review, it's slow going. However, as this is a new test, please could you put it in `fortran_gen_decls_test.py`.